backend/main/server/resources/Archive.py

```diff
@@ -0,0 +1,126 @@
+            return {'status': 'success', 'archives': archives}, 206  # Partial Content Served
+
+        return {'status': 'success', 'archives': archives}, 200
+
+    @jwt_required()
+    def post(self, who):
+        """Add archive link"""
+        if who != 'coco' and who != 'haachama':
+            return {'status': 'fail', 'message': 'No data for ' + str(who) + ' exists'}, 404
+        Archive = ArchiveCoco if who == 'coco' else ArchiveHaachama
+
+        json_data = request.get_json(force=True)
+
+        if not json_data:
+            return {'status': 'fail', 'message': 'No input data'}, 400
+
+        errors = archive_schema.validate(json_data)
+
+        if errors:
+            return {'status': 'fail', 'message': 'Error handling request'}, 422
+
+        data = archive_schema.load(json_data)
+
+        archive = Archive.query.filter_by(archiveURL=data.get('archiveURL')).first()
+
+        if archive:
+            return {'status': 'fail', 'message': 'Archive already exists'}, 400
+
+        archive = Archive(archiveURL=data.get('archiveURL'))
+
+        db.session.add(archive)
+        db.session.commit()
+
+        return {'status': 'success', 'message': 'Archive successfully created'}, 201
+
+
+class ArchiveResource(Resource):
+    @cache.cached(timeout=100)
+    def get(self, who, archiveID):
+        """"Get an archive by archive ID"""
+        if who != 'coco' and who != 'haachama':
+            return {'status': 'fail', 'message': 'No data for ' + str(who) + ' exists'}, 404
+        Archive = ArchiveCoco if who == 'coco' else ArchiveHaachama
+
+        archive = Archive.query.filter_by(archiveID=archiveID)
+
+        if not archive.first():
+            return {'status': 'fail', 'message': 'No archive with ID ' + str(archiveID) + ' exists'}, 404
+
+        archive = archives_schema.dump(archive)
+        return {'status': 'success', 'archives': archive}, 200
+
+    @jwt_required()
+    def delete(self, who, archiveID):
+        """Delete an archive by ID"""
+        if who != 'coco' and who != 'haachama':
+            return {'status': 'fail', 'message': 'No data for ' + str(who) + ' exists'}, 404
+        Archive = ArchiveCoco if who == 'coco' else ArchiveHaachama
+
+        archive = Archive.query.filter_by(archiveID=archiveID)
+
+        if not archive.first():
+            return {'status': 'fail', 'message': 'No archive with ID ' + str(archiveID) + ' exists'}, 404
+
+        archive.delete()
+        db.session.commit()
+        return {'status': 'sucess', 'message': 'Archive Deleted'}, 200
+
+
+class ArchiveRandomResource(Resource):
+    @cache.cached(timeout=60 * 60 * 24)
+    def get(self, who):
+        """"Get a random archive link"""
+        if who != 'coco' and who != 'haachama':
+            return {'status': 'fail', 'message': 'No data for ' + str(who) + ' exists'}, 404
+        Archive = ArchiveCoco if who == 'coco' else ArchiveHaachama
+
+        archiveID = random.randint(1, Archive.query.count())
```

Comment:
We can just randomly insert into the database and get rid of this random logic here. Replace this with a (date%db_size) + 1 and we can just step through the database incrementally, avoiding repeating videos